apps/condo/domains/ticket/components/TicketClassifierSelect.tsx

```diff
@@ -0,0 +1,320 @@
+    const DescriptionClassifierLabel = intl.formatMessage({ id: 'component.ticketclassifier.DescriptionLabel' })
+    const threeLvlSelectState = useRef({ id: classifierRule, place: null, category:null, description: null })
+    const client = useApolloClient()
+    const helper = new ClassifiersQueryLocal(client)
+    const validations = useTicketValidations()
+    const ticketForm = useRef(null)
+
+    const onUserSelect = (id, type) => {
+        threeLvlSelectState.current = { ...threeLvlSelectState.current, [type]: id }
+        updateLevels({ [type]: id })
+    }
+    const onUserSearch = async (input, type) => {
+        const classifiers = await helper.search(input, type)
+        Setter[type].search(classifiers)
+    }
+
+    const {
+        set: descriptionSet,
+        SelectComponent: DescriptionSelect,
+    } = useTicketClassifierSelectHook({
+        onChange: (id) => onUserSelect(id, TicketClassifierTypes.description),
+        onSearch: (id) => onUserSearch(id, TicketClassifierTypes.description),
+        initialValue: descriptionClassifier,
+    })
+
+    const {
+        set: categorySet,
+        SelectComponent: CategorySelect,
+        ref: categoryRef,
+    } = useTicketClassifierSelectHook({
+        onChange: (id) => onUserSelect(id, TicketClassifierTypes.category),
+        onSearch: (id) => onUserSearch(id, TicketClassifierTypes.category),
+        initialValue: categoryClassifier,
+    })
+
+    const {
+        set: placeSet,
+        SelectComponent: PlaceSelect,
+        ref: placeRef,
+    } = useTicketClassifierSelectHook({
+        onChange: (id) => onUserSelect(id, TicketClassifierTypes.place),
+        onSearch: (id) => onUserSearch(id, TicketClassifierTypes.place),
+        initialValue: placeClassifier,
+    })
+
+    const Setter = {
+        place: placeSet,
+        category: categorySet,
+        description: descriptionSet,
+    }
+
+    useEffect(() => {
+        helper.init().then(_ => {
+            if (threeLvlSelectState.current.id) {
+                helper.findRules({ id: threeLvlSelectState.current.id }).then(([rule]) => {
+                    const { place, category, description } = rule
+                    threeLvlSelectState.current = { ...threeLvlSelectState.current, ...{ place: place.id, category: category.id, description: description.id } }
+                    updateLevels(threeLvlSelectState.current)
+                })
+            } else {
+                helper.search('', 'place').then(places => {
+                    Setter.place.all(places)
+                })
+                helper.search('', 'category').then(categories => {
+                    Setter.category.all(categories)
+                })
+            }
+        })
+        return () => {
+            // clear all loaded data from helper
+            helper.clear()
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
remove eslint disable or rewrite hook logic

Reply:
This hook needs to be run only once on create - so it has empty deps